trade.lic

```diff
@@ -1,78 +1,425 @@
 =begin
   Documentation: https://elanthipedia.play.net/Lich_script_repository#trade
 =end
-custom_require.call(%w[common common-items common-money common-travel drinfomon equipmanager events])
+custom_require.call(%w[common common-items common-arcana common-crafting common-money common-travel drinfomon equipmanager events])
 
 class Trade
   include DRC
+  include DRCA
   include DRCT
   include DRCI
+  include DRCC
   include DRCM
 
   def initialize
     @town_data = get_data('town')
     @equipment_manager = EquipmentManager.new
     @settings = get_settings
     @contract_container = @settings.trade_contract_container
-    @hometown = @settings.hometown
-    bput("open my #{@contract_container}", 'You open', 'That is already')
-    @equipment_manager.empty_hands
+    @hub_city = 'Crossing'
+    @emergency_delivery_town = nil
+    @current_caravan_destination = nil
+    @caravan_coins_on_hand = @settings.caravan_coins_on_hand.to_i
+    @caravan_interior = @settings.caravan_interior
+    @caravan_being_led = false
+    @force_contract_from = []
+    Flags.add('lead-arrived', 'having arrived at its destination', 'You feel the momentum of the room shift and then hear a voice from outside announce')
+    Flags.add('lead-failed', 'Your caravan guide comes to you and suggests')
+    Flags.add('caravan-arrived', /^Your .*, following you/)
+
+    #TRAINING
+    @gem_pouch_adjective = @settings.gem_pouch_adjective
+    @full_pouch_container = @settings.full_pouch_container
+    @training_spells = @settings.crafting_training_spells
+    @step_toggle = -1
+    wipe_token
+    Flags.add('trading-magic-ready', 'You feel fully prepared to cast your spell.')
+
+    @equipment_manager = EquipmentManager.new(@settings)
+    @cooldown_timers = {}
+    @caravan_training_skills = @settings.caravan_training_skills
+    @lead_training_skills = @caravan_training_skills.select { |skill, _cooldown| !['Mechanical Lore', 'Augmentation', 'Utility', 'Warding'].include?(skill) }
```

Comment:
Is this the same as using .reject? without the `!`?

Reply:
Oh, yeah, I suppose it is, isn't it.  I might as well change that.